templates/webapi/test/badge.svg.ep

```diff
@@ -0,0 +1,58 @@
+% my %result_colors = (
+%	passed => '#4c1',
+%	failed => '#e05d44',
+%	parallel_failed => '#e05d44',
+%	softfailed => '#EEB560',
+%	incomplete => '#AF1E11',
+%	timeout_exceeded => '#AF1E11',
+%	blocked => '#9167b7',
+%	scheduled => '#67A2B7',
+%	cancelled => '#aaaaaa',
+%	user_cancelled => '#aaaaaa',
+%	assigned => '#8BDFFF',
+%	running => '#8BDFFF',
+%	uploading => '#7DC8E5',
+% );
+%
+% my $result = ($job->result eq 'none') ? $job->state : $job->result;
+% if ($result eq 'scheduled' && $job->blocked_by_id) {
+%	$result = 'blocked';
+% }
+% my $result_color = $result_colors{$result} // $result_colors{cancelled};
+% my $result_text = $result =~ s/_/ /rg;
+% $result_text = $job->BUILD . ': ' . $result_text if ($show_build);
+%
+% my $charlen = 7.6;
+% my $res_width = 20 + $charlen * length($result_text);
+%# now substract weighted for smaller width characters
+% $res_width -= $charlen * (() = $result_text =~ /f/g) * 0.4;
+% $res_width -= $charlen * (() = $result_text =~ /i/g) * 0.6;
+% $res_width -= $charlen * (() = $result_text =~ /j/g) * 0.5;
+% $res_width -= $charlen * (() = $result_text =~ /l/g) * 0.6;
+% $res_width -= $charlen * (() = $result_text =~ /t/g) * 0.5;
+% $res_width -= $charlen * (() = $result_text =~ /1/g) * 0.6;
+% $res_width -= $charlen * (() = $result_text =~ /\./g) * 0.7;
+% $res_width -= $charlen * (() = $result_text =~ /:/g) * 0.7;
+% my $width = $res_width + 75;
```

Comment:
However, I would get rid of all of this and let a little bit of JavaScript figure it out. That's how it would look like:

```
<svg height="20" onload="bb = document.querySelector('#text').getBBox(); this.setAttribute('width', bb.x + bb.width + 5)" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg">
  <clipPath id="a">
    <rect rx="3" width="100%" height="100%"/>
  </clipPath>
  <linearGradient id="b" x2="0" y2="100%">
    <stop offset="0" stop-color="#bbb" stop-opacity=".1"/>
    <stop offset="1" stop-opacity=".1"/>
  </linearGradient>
  <g clip-path="url(#a)">
    <rect width="100%" height="100%" fill="#555"/>
    <rect x="85" width="100%" height="100%" fill="#ff6644"/>
    <rect height="20" fill="url(#b)"/>
    <g fill="#fff" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11" id="text">
      <text x="30" y="15" fill="#010101" fill-opacity=".3">openQA</text>
      <text x="30" y="14">openQA</text>
      <text x="90" y="15" fill="#010101" fill-opacity=".3">foobar</text>
      <text x="90" y="14">foobar</text>
    </g>
  </g>
  <image height="14" width="14" x="6" xlink:href="data:image/svg+xml;…" y="3" font-family="Verdana,Geneva,DejaVu Sans,sans-serif" text-anchor="middle" text-rendering="geometricPrecision"/>
</svg>
```

I have removed the template substitution for testing purposes here so you'll have to add it back (and as mentioned in the other comment get rid of the base64-encoded part). If you want to play around with it further, you can checkout http://jsfiddle.net/db3fk61e/1/.

Reply:
By the way, this JavaScript is not black magic. It is using normal DOM-APIs and `getBBox()` which is [documented](https://developer.mozilla.org/en-US/docs/Web/API/SVGGraphicsElement/getBBox).

Reply:
The downside seems to be that only browsers seem to be able to view the svg correctly then.
eom and gimp don't seem to understand the js.

But I guess that is an issue that we can bear with.

Reply:
Maybe you want to assign a default with (e.g. `width="200"`) so it isn't completely broken without JavaScript.

Reply:
> But I guess that is an issue that we can bear with.

Yes, because it is supposed to be embedded on a web page anyways.